scripts/generate-packages.tests.ps1

```diff
@@ -0,0 +1,91 @@
+# Unit tests for the generate-packages.ps1 script.
+# Preconditions:
+# - Install chocolatey: https://docs.chocolatey.org/en-us/choco/setup/#install-with-powershellexe
+# - Install Pester: Install-Module -Name Pester -Force -SkipPublisherCheck (https://pester.dev/docs/introduction/installation#windows)
+# Running:
+# - Open a PowerShell terminal.
+# - Generate binaries: .\scripts\its-build.ps1
+# - Package the scanner: nuget pack nuspec\netcoreglobaltool\dotnet-sonarscanner.nuspec
+# - Run the tests: Invoke-Pester -Output Detailed .\scripts\generate-packages.tests.ps1
+
+BeforeAll {
+    New-Item -Force -Path "$PSScriptRoot" -Name "testcontext" -ItemType Directory
+    New-Item -Force -Path "$PSScriptRoot/testcontext" -Name "nuspec" -ItemType Directory
+    New-Item -Force -Path "$PSScriptRoot/testcontext" -Name "build" -ItemType Directory
+    New-Item -Force -Path "$PSScriptRoot/testcontext/build" -Name "sonarscanner-net.zip" -type File
+    New-Item -Force -Path "$PSScriptRoot/testcontext/build" -Name "sonarscanner-net-framework.zip" -type File
+    New-Item -Force -Path "$PSScriptRoot/testcontext/scripts/version" -Name "Version.props" -type File
+
+    Set-Location "$PSScriptRoot/testcontext"
+
+    function CheckVersion([string] $packageFileName, [string] $nuspecFileName, [string] $expectedVersion) {
+        Rename-Item -Path "build/$packageFileName" -NewName 'temp.zip'
```

Comment:
Why do you need to move/rename the file here?
Isn't this happening inside the test context?

Reply:
By default the extension is `nupkg` and `Expand-Archive` doesn't want to unpack it so I have to rename the file to zip first.